java/runtime/src/main/java/io/ray/runtime/RayNativeRuntime.java

```diff
@@ -52,124 +47,102 @@
 
-    gcsClient = new GcsClient(rayConfig.getRedisAddress(), rayConfig.redisPassword);
+      loadConfigFromGcs();
 
-    if (rayConfig.getJobId() == JobId.NIL) {
-      rayConfig.setJobId(gcsClient.nextJobId());
-    }
-    int numWorkersPerProcess =
-        rayConfig.workerMode == WorkerType.DRIVER ? 1 : rayConfig.numWorkersPerProcess;
+      if (rayConfig.getJobId() == JobId.NIL) {
+        rayConfig.setJobId(gcsClient.nextJobId());
+      }
+      int numWorkersPerProcess =
+          rayConfig.workerMode == WorkerType.DRIVER ? 1 : rayConfig.numWorkersPerProcess;
 
-    byte[] serializedJobConfig = null;
-    if (rayConfig.workerMode == WorkerType.DRIVER) {
-      JobConfig.Builder jobConfigBuilder =
-          JobConfig.newBuilder()
-              .setNumJavaWorkersPerProcess(rayConfig.numWorkersPerProcess)
-              .addAllJvmOptions(rayConfig.jvmOptionsForJavaWorker)
-              .putAllWorkerEnv(rayConfig.workerEnv)
-              .addAllCodeSearchPath(rayConfig.codeSearchPath);
-      serializedJobConfig = jobConfigBuilder.build().toByteArray();
-    }
+      byte[] serializedJobConfig = null;
+      if (rayConfig.workerMode == WorkerType.DRIVER) {
+        JobConfig.Builder jobConfigBuilder =
+            JobConfig.newBuilder()
+                .setNumJavaWorkersPerProcess(rayConfig.numWorkersPerProcess)
+                .addAllJvmOptions(rayConfig.jvmOptionsForJavaWorker)
+                .putAllWorkerEnv(rayConfig.workerEnv)
+                .addAllCodeSearchPath(rayConfig.codeSearchPath);
+        serializedJobConfig = jobConfigBuilder.build().toByteArray();
+      }
+
+      Map<String, String> rayletConfigStringMap = new HashMap<>();
+      for (Map.Entry<String, Object> entry : rayConfig.rayletConfigParameters.entrySet()) {
+        rayletConfigStringMap.put(entry.getKey(), entry.getValue().toString());
+      }
 
-    // TODO(qwang): Get object_store_socket_name and raylet_socket_name from Redis.
-    nativeInitialize(rayConfig.workerMode.getNumber(),
-        rayConfig.nodeIp, rayConfig.getNodeManagerPort(),
-        rayConfig.workerMode == WorkerType.DRIVER ? System.getProperty("user.dir") : "",
-        rayConfig.objectStoreSocketName, rayConfig.rayletSocketName,
-        (rayConfig.workerMode == WorkerType.DRIVER ? rayConfig.getJobId() : JobId.NIL).getBytes(),
-        new GcsClientOptions(rayConfig), numWorkersPerProcess,
-        rayConfig.logDir, rayConfig.rayletConfigParameters, serializedJobConfig);
-
-    taskExecutor = new NativeTaskExecutor(this);
-    workerContext = new NativeWorkerContext();
-    objectStore = new NativeObjectStore(workerContext, shutdownLock);
-    taskSubmitter = new NativeTaskSubmitter();
-
-    LOGGER.debug("RayNativeRuntime started with store {}, raylet {}",
-        rayConfig.objectStoreSocketName, rayConfig.rayletSocketName);
+      // TODO(qwang): Get object_store_socket_name and raylet_socket_name from Redis.
```

Comment:
Is this comment still valid? This should already be done in `loadConfigFromGcs`?